src/py/flwr/server/compat/driver_client_proxy_test.py

```diff
@@ -17,348 +17,277 @@
+        error_reply: bool = False,
+    ) -> Callable[[Iterable[Message]], Iterable[str]]:
+        """Get the push_messages function that sets the return value of pull_messages
+        when called."""
+
+        def push_messages(messages: Iterable[Message]) -> Iterable[str]:
+            msg = list(messages)[0]
+            if error_reply:
+                recordset = None
+                ret = msg.create_error_reply(ERROR_REPLY)
+            elif isinstance(res, GetParametersRes):
+                recordset = compat.getparametersres_to_recordset(res, True)
+            elif isinstance(res, GetPropertiesRes):
+                recordset = compat.getpropertiesres_to_recordset(res)
+            elif isinstance(res, FitRes):
+                recordset = compat.fitres_to_recordset(res, True)
+            elif isinstance(res, EvaluateRes):
+                recordset = compat.evaluateres_to_recordset(res)
+            else:
+                raise ValueError(f"Unsupported type: {type(res)}")
+            if recordset is not None:
+                ret = msg.create_reply(recordset)
+            ret.metadata.__dict__["_message_id"] = REPLY_MESSAGE_ID
+
+            # Set the return value of `pull_messages`
+            self.driver.pull_messages.return_value = [ret]
+            return [INSTRUCTION_MESSAGE_ID]
+
+        return push_messages
+
+    def _common_assertions(self, original_ins: Any) -> None:
+        """Check common assertions."""
+        # Check if the created message contains the orignal *Ins
+        assert self.created_msg is not None
+        actual_ins = {  # type: ignore
+            GetPropertiesIns: compat.recordset_to_getpropertiesins,
+            GetParametersIns: compat.recordset_to_getparametersins,
+            FitIns: (lambda x: compat.recordset_to_fitins(x, True)),
+            EvaluateIns: (lambda x: compat.recordset_to_evaluateins(x, True)),
+        }[type(original_ins)](self.created_msg.content)
+        self.assertEqual(self.called_times, 1)
+        self.assertEqual(actual_ins, original_ins)
+
+        # Check if push_messages is called once with expected args/kwargs.
+        self.driver.push_messages.assert_called_once()
+        try:
+            self.driver.push_messages.assert_any_call([self.created_msg])
+        except AssertionError:
+            self.driver.push_messages.assert_any_call(messages=[self.created_msg])
 
-    def __init__(self, *args, **kwargs) -> None:  # type: ignore
-        """Prepare Tests with GrpcDriver."""
-        super().__init__(GrpcDriver(), *args, **kwargs)
+        # Check if pull_messages is called once with expected args/kwargs.
+        self.driver.pull_messages.assert_called_once()
+        try:
+            self.driver.pull_messages.assert_called_with([INSTRUCTION_MESSAGE_ID])
+        except AssertionError:
+            self.driver.pull_messages.assert_called_with(
+                message_ids=[INSTRUCTION_MESSAGE_ID]
+            )
```

Comment:
This exception is never triggered.

Reply:
This is not triggered for now, but it would better to keep it, in case that an implementation of `DriverClientProxy` calls `driver.pull_messages(message_ids=[...])` instead of `driver.pull_messages([...])`. I think we shouldn't assume how `DriverClientProxy` calls the Driver methods in the unit tests.